lib/semian/grpc.rb

```diff
@@ -26,14 +26,40 @@ module GRPC
     ResourceBusyError = ::GRPC::ResourceBusyError
     CircuitOpenError = ::GRPC::CircuitOpenError
 
-    def initialize(host, creds, **opts)
-      @raw_semian_options = opts[:semian_options]
-      opts.delete(:semian_options)
-      super(host, creds, opts)
+    class SemianConfigurationChangedError < RuntimeError
+      def initialize(msg = "Cannot re-initialize semian_configuration")
+        super
+      end
+    end
+
+    class << self
+      attr_accessor :exceptions
+      attr_reader :semian_configuration
+
+      def semian_configuration=(configuration)
+        raise Semian::GRPC::SemianConfigurationChangedError unless @semian_configuration.nil?
+        @semian_configuration = configuration
+      end
+
+      def retrieve_semian_configuration(host)
+        @semian_configuration.call(host) if @semian_configuration.respond_to?(:call)
+      end
+    end
+
+    def raw_semian_options
+      @raw_semian_options ||= begin
+        if @host.empty?
+          host = @ch.target
```

Comment:
I'm finding this stuff pretty confusing. I have no idea what `@ch` is for example. Why do we go there to get the host? Why is `@host` not present? Could we add a comment?